### ISSUE TITLE
`cleos system delegatebw` in some case requires the auth of the receiver

### ISSUE THREAD

**Issue by andriantolie** [EOSIO/eos#3630]
So in this case I have 2 accounts called andrianto and andri. Details are the following
```
$ cleos get account andrianto
privileged: false
permissions:
     owner     1:    1 EOS8kE63z4NcZatvVWY4jxYdtLg6UEA123raMGwS6QDKwpQ69eGcP
        active     1:    1 EOS8kE63z4NcZatvVWY4jxYdtLg6UEA123raMGwS6QDKwpQ69eGcP
memory:
     quota:     60.32 Mb     used:     4.016 Kb

net bandwidth: (averaged over 3 days)
     staked:   25000150.0000 SYS           (total stake delegated from account to self)
     delegated:      10.0000 SYS           (total staked delegated to account from others)
     used:             6.832 Kb
     available:        13.04 Tb
     limit:            13.04 Tb

cpu bandwidth: (averaged over 3 days)
     staked:   25000150.0000 SYS           (total stake delegated from account to self)
     delegated:      10.0000 SYS           (total staked delegated to account from others)
     used:             71.98 ms
     available:    3.982e+05 hr
     limit:        3.982e+05 hr

producers:     <not voted>
```
and 
```
cleos get account andri
privileged: false
permissions:
     owner     1:    1 EOS6MRyAjQq8ud7hVNYcfnVPJqcVpscN5So8BhtHuGYqET5GDW5CV
        active     1:    1 EOS6MRyAjQq8ud7hVNYcfnVPJqcVpscN5So8BhtHuGYqET5GDW5CV
memory:
     quota:     59.44 Mb     used:     3.498 Kb

net bandwidth: (averaged over 3 days)
     staked:       1000.0000 SYS           (total stake delegated from account to self)
     delegated: 1000010.0000 SYS           (total staked delegated to account from others)
     used:               810 bytes
     available:        534.6 Gb
     limit:            534.6 Gb

cpu bandwidth: (averaged over 3 days)
     staked:       1000.0000 SYS           (total stake delegated from account to self)
     delegated: 1000010.0000 SYS           (total staked delegated to account from others)
     used:             8.857 ms
     available:    1.594e+04 hr
     limit:        1.594e+04 hr

producers:
     andrianto
```

Calling `cleos system delegatebw eosio andrianto "10.0000 SYS" "10.0000 SYS" --transfer` success without any problem. However, when I tried to call `cleos system delegatebw eosio andri "10.0000 SYS" "10.0000 SYS" --transfer`, it fails with the following error
<img width="1047" alt="screen shot 2018-05-30 at 6 10 20 pm" src="https://user-images.githubusercontent.com/6541441/40751783-af20d2f2-643a-11e8-8e4f-8293485ca40b.png">
Saying that andri auth is required which is kinda weird since andri is the receiver in this case.

Without --transfer, everything is fine

My wallet has the private keys of eosio, andrianto, and andri.

**Comment by andriantolie** [comment]
Strangely this is fine in local node but not in the testnet

**Comment by andriantolie** [comment]
Okay. I think I got the condition to trigger this.
For the above condition, it seems that andrianto didn't cast a vote while andri cast a vote. When andri removed his vote, the above (delegatebw andri) action can be executed successfully. While when andrianto cast his vote, the above (delegatebw andrianto) action fails.

This explains why in my local node, I didn't see that problem since I didn't cast a vote yet...

**Comment by wanderingbort** [comment]
resolved by #3679
